### PR TITLE
Coupon expiration and item disabled checks

### PR DESCRIPTION
**Pull request by forellana-digithinkit**
- Fixes issue where expired coupons were kept around during checkout
- Fixes issue where dissabled items were kept in carts during checkout
- Added awc cache clear when an item is changed

You can test this by:
- Create coupon
- Apply to cart
- Set expiration datetime to some time in the past and save
- Reload your cart, the coupon should be removed.

expired-coupon-fix-aws

### CODE REVIEW

File: awesome_cart/hooks.py [hunk 1]
```diff
@@ -55,6 +55,9 @@
 	},
 	"Sales Invoice": {
 		"validate": ["awesome_cart.utils.quotation_validate"]
+	},
+	"Item": {
+		"on_change": ["awesome_cart.utils.clear_cache_on_doc_update"]
```
Comment (vjFaLk) [inline awesome_cart/hooks.py]: You wanna do this on_update instead, I think on_change is broken or something

**Comment by Alchez** [comment]
I had a coupon that was applicable on all CIEMs and UIEMs, and this happened:

image

**Comment by Alchez** [comment]
Also:

- After applying the coupon, I changed the items that the discount was applicable on, but the discount value did not change on refresh.
- Not sure if a big deal, but if the discount is applied on Net Total, a free-shipping + item discount coupon ends up applying the shipping discount on the items instead.

**Comment by Alchez** [comment]
Rendering looks fine, but I noticed some issues in how item discounts are applied in AWC Coupon.

<hr>

If you select **Discount Type** as **Value Discount** and give it a value amount, it seems to apply the discount on a single quantity of the item, even though it says it's been applied to all of them.

**Example:** (the coupon applies a $250 value discount)

image

<hr>

If you select **Discount Type** as **Actual Value** and give it a value amount, the coupon splits the value based on the number of items, and proportionately applies the discount.

**Example:** (the coupon applies a $250 actual value discount, but each quantity receives a $125 discount)
image

**Comment by forellana-digithinkit** [comment]
As per last conversation. I'm merging this. Remaining issues not part of this PR will be made into issues.

**Pull request merged**